File: LBExpandFriendList/LBExpandFriendList/home/LBHomeCell.m
//
//  LBHomeCell.m
//  LBExpandFriendList
//
//  Created by yunmei on 2018/1/20.
//  Copyright © 2018年 yunmei. All rights reserved.
//

#import "LBHomeCell.h"
#import "Masonry.h"
@implementation LBHomeCell{
    UIView *_downLine;
}

+ (instancetype)cellWithTableView:(UITableView *)tableview{
    
    LBHomeCell *cell = [tableview dequeueReusableCellWithIdentifier:@"LBHomeCell"];
    if (!cell) {
        cell = [[LBHomeCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"LBHomeCell"];
    }
    tableview.separatorStyle = UITableViewCellSeparatorStyleNone;
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}

-(void)awakeFromNib{
    [super awakeFromNib];
}
-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        
        [self setUpLayout];
    }
    return  self;
}
- (void)setUpLayout{
    /// 标题
    _titleLabel = [[UILabel alloc]init];
    _titleLabel.font = [UIFont systemFontOfSize:15];
    [self.contentView addSubview:_titleLabel];
    __weak typeof(self.contentView) weakSelf = self.contentView;
    [_titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.center.equalTo(weakSelf);
    }];
    /// 底部的线
    _downLine = [[UIView alloc]init];
    _downLine.backgroundColor = [UIColor grayColor];
    [self.contentView addSubview:_downLine];
    [_downLine mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.mas_equalTo(weakSelf.mas_bottom);
        make.height.equalTo(@1);
        make.left.right.equalTo(weakSelf);
    }];
}

@end

File: LBSwiftExpandFriendList/LBSwiftExpandFriendList/home/LBHomeCell.swift
//
//  LBHomeCell.swift
//  LBSwiftExpandFriendList
//
//  Created by yunmei on 2018/1/20.
//  Copyright © 2018年 liubo. All rights reserved.
//

import UIKit
import SnapKit
class LBHomeCell: UITableViewCell {
    var titleLabel:UILabel = UILabel()
    var downLine:UIView = UIView()
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        setUpLayout()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}

extension LBHomeCell{
    func setUpLayout(){
        /// 标题
        titleLabel.font = UIFont.systemFont(ofSize: 15)
        contentView.addSubview(titleLabel)
        titleLabel.snp.makeConstraints { (make) in
            make.center.equalTo(contentView);
        }
        /// 底部的线
        downLine.backgroundColor = UIColor.gray
        contentView.addSubview(downLine)
        downLine.snp.makeConstraints { (make) in
            make.bottom.equalTo(contentView.snp.bottom);
            make.height.equalTo(1);
            make.left.right.equalTo(contentView);
        }
    }
}
